pupil-spa/src/app/services/answer/answer.service.ts

```diff
@@ -15,8 +18,33 @@ export class AnswerService {
       answers = [];
     }
 
+    // Check for duplicates - we can only store one answer for each question
+    const previousAnswers = this.getPreviousAnswers(answer, answers);
+
+    if (previousAnswers.length > 0) {
+      // Log this error
+      this.auditService.addEntry(new DuplicateAnswerError(answer));
+      return;
+    }
     answers.push(answer);
     this.storageService.setItem('answers', answers);
   }
 
+  /**
+   * Find a previous answer in the answers array
+   * @param {Answer} answer - the current answer to be stored
+   * @param [Answer] answers - the list of previously stored answers
+   * @return [Answer?] - empty array or array of answers
+   */
+  private getPreviousAnswers(answer: Answer, answers) {
+    return answers.filter(
+      x => {
```

Comment:
Isn't it more readable like this? `x => x.sequenceNumber === answer.sequenceNumber && x.factor1 === answer.factor1 && x.factor2 === answer.factor2`
You are basically using an if to return boolean

Reply:
@GeorgeChatzigiannis - I thought it was less readable that way, sorry.

Reply:
I guess readability can be subjective

Reply:
@GeorgeChatzigiannis - happy to oblige if there is sufficient support for the change

Reply:
@jon-shipley it's a minor detail really if everyone else approves it as it stands I am happy.